release-20.2: pgwire: error cleanly when a simple query is too big

otan opened the pull request:
Backport 1/1 commits from #53790.

/cc @cockroachdb/release

---

Refs: #50330

Release justification: low risk, high benefit changes to
existing functionality

Release note (sql change): Previously, certain SQL commands
sent over the PostgreSQL FE/BE protocol to which were too big would
error opaquely. This is now resolved for non-PREPARE related statements,
clearly error messaging instead.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54067)
<!-- Reviewable:end -->

otan committed:
pgwire: error cleanly when a simple query is too big

Release justification: low risk, high benefit changes to
existing functionality

Release note (sql change): Previously, certain SQL commands
sent over the PostgreSQL FE/BE protocol to which were too big would
error opaquely. This is now resolved for non-PREPARE related statements,
clearly error messaging instead.
